File: android/nativepreview/src/main/java/com/travelplaner/nativepreview/domain/SavedPlaceDocument.kt
package com.travelplaner.nativepreview.domain
import kotlinx.serialization.json.*

data class SavedPlaceDocument(val id: String, val sourceKey: String, val payload: JsonObject, val createdAt: Long, val updatedAt: Long) {
    val draft get() = PlaceDraft.from(payload)
    // A saved row is already a durable user-owned record. Copy it as an
    // existing payload so extension fields survive a second itinerary copy;
    // accepting a fresh SDK result remains the Google selection path.
    val selection: PlaceSelection get() = PlaceSelection(PlaceReference.Existing,originalItem=payload,sourceSavedID=id)
}

File: ios/NativeCore/Sources/NativeCore/SavedPlaceDocument.swift
import Foundation

public struct SavedPlaceDocument: Identifiable {
    public let id, sourceKey: String
    public let payload: [String: Any]
    public let createdAt, updatedAt: Double
    public init(id: String, sourceKey: String, payload: [String: Any], createdAt: Double, updatedAt: Double) {
        self.id = id; self.sourceKey = sourceKey; self.payload = payload; self.createdAt = createdAt; self.updatedAt = updatedAt
    }
    public var draft: PlaceDraft { PlaceDraft(item: payload) }
    public var selection: PlaceSelection {
        // A saved row is already a durable user-owned record. Copy it as an
        // existing payload so extension fields survive a second itinerary copy;
        // accepting a fresh SDK result remains the Google selection path.
        return .init(reference: .existing, originalItem: payload, sourceSavedID: id)
    }
}
